latch/registry/table.py

```diff
@@ -0,0 +1,275 @@
+                                nodes {
+                                    data
+                                    key
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            """,
+            variables={"argTableId": self.id},
+        )
+
+        return self._info
+
+    def get_display_name(self, load_if_missing=False):
+        if not load_if_missing or self._name is not None:
+            return self._name
+
+        self._name = self._get_info()["catalogExperiment"]["displayName"]
+
+        return self._name
+
+    def get_columns(self, load_if_missing=False):
+        if not load_if_missing or self._columns is not None:
+            return self._columns
+
+        self._columns = self._get_info()["catalogExperiment"][
+            "catalogExperimentColumnDefinitionsByExperimentId"
+        ]["nodes"]
+
+        return self._columns
+
+    def list_records(self) -> Iterator[ListRecordsOutputType]:
+        has_next_page = True
+        end_cursor = None
+
+        while has_next_page:
+            data = execute(
+                """
+                query tableQuery($argTableId: BigInt!, $argAfter: Cursor) {
+                    catalogExperiment(id: $argTableId) {
+                        catalogSamplesByExperimentId(
+                            condition: { removed: false }
+                            first: 50
+                            after: $argAfter
+                        ) {
+                            nodes {
+                                id
+                                name
+                                catalogSampleColumnDataBySampleId {
+                                    nodes {
+                                        data
+                                        key
+                                    }
+                                }
+                            }
+                            pageInfo {
+                                endCursor
+                                hasNextPage
+                            }
+                        }
+                    }
+                }
+                """,
+                {"argTableId": self.id, "argAfter": end_cursor},
+            )["catalogExperiment"]["catalogSamplesByExperimentId"]
+
+            records = data["nodes"]
+            end_cursor = data["pageInfo"]["endCursor"]
+            has_next_page = data["pageInfo"]["hasNextPage"]
+
+            output: List[Record] = []
+            errors: List[Exception] = []
+
+            for record in records:
+                record_data = record["catalogSampleColumnDataBySampleId"]["nodes"]
+
+                values = {}
+
+                try:
+                    for column in self.get_columns(load_if_missing=True):
+                        key = column["key"]
+                        typ = column["type"]
+
+                        for data_point in record_data:
```

Comment:
`O(n^2)` algorithm where `n` is the number of columns

use a dict over columns here to turn it into `O(n)`